sound/soc/sof/topology.c

```diff
@@ -571,7 +571,11 @@ static int sof_copy_tuples(struct snd_sof_dev *sdev, struct snd_soc_tplg_vendor_
 						continue;
 
 					tuples[*num_copied_tuples].token = tokens[j].token;
-					tuples[*num_copied_tuples].value.s = elem->string;
```

Comment:
I'd change the commit message to say: "In preparation to handle tuples from multiple topologies, duplicate the tuple string value by allocating memory for it..."